pstoremanager: test timeout

```
panic: test timed out after 10m0s
  
  goroutine 50 [running]:
  testing.(*M).startAlarm.func1()
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1788 +0x8e
  created by time.goFunc
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/time/sleep.go:180 +0x31
  
  goroutine 1 [chan receive, 9 minutes]:
  testing.tRunner.func1()
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1225 +0x311
  testing.tRunner(0xc000102b60, 0xc0001d1ce0)
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1265 +0x13b
  testing.runTests(0xc00099a280, {0x5356b40, 0x3, 0x3}, {0x408c0ad, 0x1a, 0x55e34e0})
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1596 +0x43f
  testing.(*M).Run(0xc00099a280)
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1504 +0x51d
  main.main()
  	_testmain.go:651 +0x20a
  
  goroutine 42 [chan receive, 10 minutes]:
  github.com/libp2p/go-libp2p/p2p/host/pstoremanager_test.TestClose(0xc000102d00)
  	/Users/runner/work/go-libp2p/go-libp2p/p2p/host/pstoremanager/pstoremanager_test.go:97 +0x3c5
  testing.tRunner(0xc000103[380](https://github.com/libp2p/go-libp2p/runs/6681155337?check_suite_focus=true#step:7:382), 0x4bb25f0)
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1259 +0x102
  created by testing.(*T).Run
  	/Users/runner/hostedtoolcache/go/1.17.10/x64/src/testing/testing.go:1306 +0x35a
  FAIL	github.com/libp2p/go-libp2p/p2p/host/pstoremanager	600.587s

--- a/p2p/host/pstoremanager/pstoremanager_test.go
+++ b/p2p/host/pstoremanager/pstoremanager_test.go
@@ -86,11 +86,18 @@ func TestClose(t *testing.T) {
 
 	emitter, err := eventBus.Emitter(new(event.EvtPeerConnectednessChanged))
 	require.NoError(t, err)
+
+	sub, err := eventBus.Subscribe(&event.EvtPeerConnectednessChanged{})
+	require.NoError(t, err)
+
 	require.NoError(t, emitter.Emit(event.EvtPeerConnectednessChanged{
 		Peer:          "foobar",
 		Connectedness: network.NotConnected,
 	}))
-	time.Sleep(10 * time.Millisecond) // make sure the event is sent before we close
+
+	// make sure the event is sent before we close
+	<-sub.Out()
+
 	done := make(chan struct{})
 	pstore.EXPECT().RemovePeer(peer.ID("foobar")).Do(func(peer.ID) { close(done) })
 	require.NoError(t, man.Close())

--- a/p2p/host/pstoremanager/pstoremanager_test.go
+++ b/p2p/host/pstoremanager/pstoremanager_test.go
@@ -96,10 +96,18 @@ func TestClose(t *testing.T) {
 	}))
 
 	// make sure the event is sent before we close
-	<-sub.Out()
+	select {
+	case <-sub.Out():
+	case <-time.After(5 * time.Second):
+		t.Fatalf("Hit timeout")
+	}
 
 	done := make(chan struct{})
 	pstore.EXPECT().RemovePeer(peer.ID("foobar")).Do(func(peer.ID) { close(done) })
 	require.NoError(t, man.Close())
-	<-done
+	select {
+	case <-done:
+	case <-time.After(5 * time.Second):
+		t.Fatalf("Hit timeout")
+	}
 }

